### ISSUE TITLE
Resuming of pipeline is non ideal

### ISSUE THREAD

**Issue by photocyte** [transXpress/transXpress-nextflow#11]
The pipeline currently will only "-resume" up to the completion of the TrinityChrysalis step, which is non ideal, as the rest of the pipeline after that point is very computationally intensive & shouldn't typically need to be redone once the pipeline is completed.

I expect the failure to resume has something to do with the use of the TrinityWorkDir, especially that the Butterfly steps seem to write to this directory, without it being tracked by the process?

I've tried a few things, like putting the "cache deep" directive in several steps, adding a discrete process for the "collecting" of the butterfly parallel processes (rather than this collection being in the script outside of a process), but the pipeline still doesn't properly resume.

I've also tried putting more logic into the production of the trinityParallelCmds, so the process would be aware of what files it was working with, but never was able to fully get it working:
```
///Parses the .sh file written by Trinity to get the path of the relevant files, so the process can be aware of them
process parseDistributedCommands {
   echo true
   input:
    val parallelCommand from trinityParallelCmdsValues
   output:
    "$trinityButterflyInputFiles_ch" into trinityButterflyInputFiles_ch2
    "$trinityButterflyOutputFiles_ch" into trinityButterflyOutputFiles_ch2
   exec:
    cmds = file(parallelCommand).text.split("\n") //All the commands, separated into a list
    expectedInput=[]
    expectedOutput=[]
    cmds.each { expectedInput.add(it.split(" ")[2]); expectedOutput.add(it.split(" ")[4]) } //Iterate over each command, and split into components
    trinityButterflyInputFiles_ch = Channel.fromPath(expectedInput)
    trinityButterflyOutputFiles_ch = Channel.fromPath(expectedOutput)
}

trinityCmds.splitText(by: 10, file: "trinityCmd").map{ [it,Channel.FromPath(parseTrinityCmd(it,2)),Channel.FromPath(parseTrinityCmd(it,4))] }.into { trinityParallelCmds; trinityParallelCmdsValues}
//Then can feed the tuple to the next process and correctly set the input / output files?
```

So, it is not obvious to me if there is a solution easier than redoing all the earlier Trinity pipeline logic in Nextflow, so that the processes could be aware of all the relevant file manipulations

**Comment by photocyte** [comment]
This branch has some development work on "relativizing" the trinityCmds to get around this issue.  On the whole not so straightforward though:
https://github.com/photocyte/transXpress/tree/trinity_relative

Alternatively, could run Trinity in a single execution, rather than stepwise.  Is it really so valuable that the Nextflow pipeline of reentrancy of the discrete Trinity steps?

**Comment by photocyte** [comment]
Fixed with c9dd89b77fa030705ed9d46ada86a2bb8ebeb3ba
